src/components/userinfo.jsx

```diff
@@ -0,0 +1,29 @@
+import React, { useContext } from 'react'
+import userContext, { UserConsumer } from '../app/usercontext'
+
+const UserInfo = () => {
+  
+  const { users } = useContext(userContext)
+
+  const loggedin = users.user.loggedin
+
+
+return(
+  
+<UserConsumer>
+        {value => 
+        
+        <>
+
+            <h2>{loggedin?value.users.user.info.firstname:null}</h2>
```

Comment:
You can do it more clean like:
optional A:
```
{loggedin
 ?(<h2>{value.users.user.info.firstname}</h2>)
 : null
}
```
option B:
assign every row into a variable and then use those variables in the place